codec/websocket/stream.go

```diff
@@ -820,6 +830,10 @@ func (s *WebsocketStream) upgrade(
 	}
 	s.hb = s.hb[:0]
 
+	if s.upResCb != nil {
```

Comment:
shouldn't this be called after we assert that the handshake response is valid below? Or is there a use-case to snoop into a potentially invalid response?

Reply:
The latter. The main motivation for this is to be able to log the request/response so we can debug failed handshakes.